src/main/java/frc/robot/subsystems/Led.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.subsystems;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.subsystems.LedMap;
+
+public class Led extends SmartSubsystemBase {
+
+    LedMap map;
+    AddressableLED led;
+    // Reuse buffer
+    // Default to a length of 60, start empty output
+    // Length is expensive to set, so only set it once, then just update data
+    AddressableLEDBuffer ledBuffer;
+
+    public Led(LedMap map) {
+        led = map.led;
+        ledBuffer = map.ledBuffer;
+
+        this.map = map;
+        // PWM port 9
+        // Must be a PWM header, not MXP or DIO
+
+        led.setLength(ledBuffer.getLength());
+        led.start();
+
+    }
+
+    public void setColor(int r, int g, int b) {
+        for (var i = 0; i < ledBuffer.getLength(); i++) {
+            // Sets the specified LED to the RGB values for red
+            ledBuffer.setRGB(i, r, g, b);
+        }
+        led.setData(ledBuffer);
+    }
+
+    public CommandBase colorAlliance() {
+        return cmd("Set to Alliance Color").onExecute(() -> {
```

Comment:
Should this be `onExecute`? Once it sets colors, the LEDs will stay on those colors until they're changed.